crates/e2e/src/setup/solver.rs

```diff
@@ -0,0 +1,175 @@
+//! Mock solver for testing purposes. It returns a custom solution.
+
+use {
+    app_data::AppDataHash,
+    axum::Json,
+    ethcontract::common::abi::ethereum_types::Address,
+    model::{
+        order::{BuyTokenDestination, OrderData, OrderKind, SellTokenSource},
+        signature::EcdsaSigningScheme,
+        DomainSeparator,
+    },
+    reqwest::Url,
+    solvers_dto::{
+        auction::Auction,
+        solution::{Asset, Kind, Solution, Solutions},
+    },
+    std::sync::{Arc, Mutex},
```

Comment:
@MartinquaXD oh I see, apparently the improved that, normal mutexes used to block tokio threads time ago, I lived that sadly :sweat: ok in that case let's leave it like this for the test :ok_hand:

Reply:
The regular Mutexes still do block the whole thread. It's just that there is now a clippy lint that complains if you hold a regular mutex over an `.await` so the issue is mostly mitigated in our CI.
You are still screwed if your mutex gets locked too long (couple ms or more) but since we usually have relatively little contention on mutexes and mostly clone stuff out of it we can usually use regular mutexes just fine.